ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -700,13 +700,12 @@ def insert(
 
             from_table_expr = obj
 
-            with self.begin() as bind:
-                if from_table_expr is not None:
-                    compiled = from_table_expr.compile()
-                    columns = [
-                        self.con.dialect.normalize_name(c)
-                        for c in from_table_expr.columns
-                    ]
+            if from_table_expr is not None:
+                compiled = from_table_expr.compile()
```

Comment:
With these changes `compile` now potentially calls `self.current_database` (only for `ops.DatabaseTable`s, so we know we have a connection; `ops.UnboundTable`s are handled elsewhere), which would introduce nested transactions and those aren't supported in DuckDB it seems. So, I moved the transaction such that it's no longer nested.